### PR TITLE
Docedex will now store patients assigned to doctors

### LINKED ISSUE #67: Update storage to store patients

### PR DESCRIPTION
**Pull request by praveenkrishna0512**
**What's been done**
- Storage Components have been amended to support the storage of patients assigned to each doctor.

This PR will close #67.

### CODE REVIEW

**Comment by praveenkrishna0512** [comment]
Do not merge yet, pushing amends to update patient list

**Comment by praveenkrishna0512** [comment]
Changes complete, please review and merge, thank you!

File: src/main/java/seedu/address/storage/JsonAdaptedPatient.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        }
+        if (!Remark.isValidRemark(remark)) {
+            throw new IllegalValueException(Remark.MESSAGE_CONSTRAINTS);
+        }
+        final Remark modelRemark = new Remark(remark);
+
+        Person doctorPerson = super.toModelType();
+        return new Patient(doctorPerson.getName(), doctorPerson.getPhone(),
+                doctorPerson.getEmail(), modelHeight, modelWeight,
+                modelDiagnosis, modelStatus, modelRemark, doctorPerson.getTags());
```
Comment (Bobfree546) [inline src/main/java/seedu/address/storage/JsonAdaptedPatient.java:113]: `doctorPerson` should be `patientPerson`

**Review by Bobfree546 — CHANGES_REQUESTED** [review]
Just one change required

File: src/main/java/seedu/address/storage/JsonAdaptedPatient.java (hunk 1 above)
Comment (praveenkrishna0512) [reply src/main/java/seedu/address/storage/JsonAdaptedPatient.java]: Change made!

**Review by Bobfree546 — APPROVED** [review]
LGTM

**Pull request merged**